### PR TITLE
GB-5006 - Add federation to SDK

### PR DESCRIPTION
**Pull request by yoav-lavi**
# Description

## Features

- Allows setting the federation version in the SDK

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: packages/grafbase-sdk/src/federation.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export interface FederationParams {
+  version: string
+}
+
+export class Federation {
+  private version: string
```
Comment (obmarg) [inline packages/grafbase-sdk/src/federation.ts:6]: Wonder if it's worth typing this more specifically - we only accept the string `2.3` currently, so I _think_ we could give users faster feedback with the type system?

**Review by obmarg — APPROVED** [review]
Good stuff, thanks for helping with this

File: packages/grafbase-sdk/src/federation.ts (hunk 1 above)
Comment (yoav-lavi) [reply packages/grafbase-sdk/src/federation.ts:6]: Will add that as a specific value then

Comment (yoav-lavi) [reply packages/grafbase-sdk/src/federation.ts:6]: Done

File: packages/grafbase-sdk/tests/unit/federation.test.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      federation: {
+        version: '2.3'
+      }
+    })
+
+    expect(renderGraphQL(cfg)).toMatchInlineSnapshot(`
+      "type Post {
+        id: ID!
+      }
+      extend schema @federation(version: "2.3")
```
Comment (pimeys) [inline packages/grafbase-sdk/tests/unit/federation.test.ts]: The extensions should maybe happen before type declarations, as we do with other extensions such as the cache.

File: packages/grafbase-sdk/tests/unit/federation.test.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      "
+    `)
+  })
+
+  it('does not render federation when disabled', async () => {
```
Comment (pimeys) [inline packages/grafbase-sdk/tests/unit/federation.test.ts:28]: Is this necessary? We have plenty of tests where we don't have federation set and they should show breakage.

**Review by pimeys — APPROVED** [review]

File: packages/grafbase-sdk/tests/unit/federation.test.ts (hunk 3 above)
Comment (yoav-lavi) [reply packages/grafbase-sdk/tests/unit/federation.test.ts:28]: Fair enough

Comment (yoav-lavi) [reply packages/grafbase-sdk/tests/unit/federation.test.ts]: Removed

File: packages/grafbase-sdk/tests/unit/federation.test.ts (hunk 2 above)
Comment (yoav-lavi) [reply packages/grafbase-sdk/tests/unit/federation.test.ts]: Done

**Pull request merged**